config/spinnaker.yml

```diff
@@ -323,12 +323,28 @@ providers:
       address: https://index.docker.io/
       repository: library/nginx
 
+openstack:
+    # This default configuration uses the same environment variable names set in
+    # the OpenStack RC file. See
+    # http://docs.openstack.org/user-guide/common/cli-set-environment-variables-using-openstack-rc.html
+    # for details on the OpenStack RC file.
+    enabled: false
+    primaryCredentials:
+      name: my-openstack-account
+      authUrl: ${OS_AUTH_URL}
+      username: ${OS_USERNAME}
```

Comment:
On one hand I see this seems to be standard environment variable naming for open stack based on the link you provided. So this makes sense when you are running on openstack. I'm not as sure if this is ideal if you were running elsewhere and deploying into openstack. I guess it is ok for now -- it's easy enough to resolve and seems reasonable when deploying on openstack which is your primary concern. I'm just asking.

I'm still not entirely sure about how the password/username work. Where are they set in practice? In the other environment variable precedents here, they are set in /etc/default/spinnaker. However that file is not protected (it is 644 where passwords would suggest no more than 600). We'll have to take some additional actions to protect that file. Or do this come from some other user configuration where these variables are set outside the scope of spinnaker?